### DISCUSSION TITLE
请问可以在安卓上运行吗

### DISCUSSION THREAD

**Discussion by chen2021-web** [SeaHOH/GotoX#199] [问答/咨询] [answered]

**Comment by SeaHOH** [comment, marked as answer]
可以用 Termux 安装，但是现在很多 App 都有自己的证书，恐怕不会太有用。

**Reply by SeaHOH** [reply]
具体安装大概是这样：

1. 安装好 Termux 并打开，输入以下命令安装所需软件
    ```console
    pkg update
    pkg upgrade
    pkg install binutils clang rust git vim python
    export CARGO_BUILD_TARGET=aarch64-linux-android
    ```
1. 下载 GotoX 并安装依赖
    ```console
    git clone https://github.com/SeaHOH/GotoX.git
    cd GotoX
    pip install -r requirements.txt
    ```
1. 编辑配置文件
    ```console
    vim ./config/Config.ini
    ```
1. 设置运行权限
    ```console
    mv start.py start && chmod +x start
    ```

1. 运行
    ```console
    ./start
    ```

**Reply by 722tzz** [reply]
请教如何导入证书
我的手机是miui13，浏览器为Chrome
在按照前述方法在Termux 中安装完成后，成功启动，并使用设置代理成功，能够访问gotox.go

![IMG_20220608_104432](https://user-images.githubusercontent.com/12823558/172525237-9e84cc43-3299-45c5-97ff-e6a848ce4e1a.jpg)

下载后提示在设定中安装证书

![IMG_20220608_104507](https://user-images.githubusercontent.com/12823558/172525267-38e4a0c2-00c9-404c-a71d-360177958faa.jpg)

在设置中搜索证书得到如下结果

![IMG_20220608_104208](https://user-images.githubusercontent.com/12823558/172525939-6e40035f-91f2-4e21-8faa-b350aad12449.jpg)

选择CA证书，进入后导入如下两个证书

![IMG_20220608_104542](https://user-images.githubusercontent.com/12823558/172525316-9c609c46-0ab3-4826-a970-24ac1a163c90.jpg)

显示证书已安装

![IMG_20220608_104637](https://user-images.githubusercontent.com/12823558/172525327-e6618d8f-dfa5-4df4-bb0d-8aa1af53bc48.jpg)

但是访问google时仍然报400

![IMG_20220608_104744](https://user-images.githubusercontent.com/12823558/172525333-75a56d4e-7412-4e94-9e25-c6a2c259d0fa.jpg)

Termux 中报错如下

![IMG_20220608_111327](https://user-images.githubusercontent.com/12823558/172525347-96f34f24-77bf-42dd-bfb9-4f777e24d5f4.jpg)

不知道是哪步做的不对
谢谢！

**Reply by SeaHOH** [reply]
步骤好像都对，你看看“用户凭据”中是否有显示已安装的证书。

**Reply by 722tzz** [reply]
能看到一个这个，但是实际导入了GotoXCA.crt和ca.cer.crt两个
![image](https://user-images.githubusercontent.com/12823558/172844647-b124faca-1f2d-4619-a8d1-8b7c0378b07d.png)

**Reply by SeaHOH** [reply]
这两个链接是相同的文件，只是文件描述不同，前一个是安装，后一个是下载。
既然已安装正确，那么就是最开头提到的：
> 但是现在很多 App 都有自己的证书，恐怕不会太有用。

**Comment by chen2021-web** [comment]
pip install -r requirements.txt这个命令报错

**Comment by chen2021-web** [comment]
ERROR: Command errored out with exit status 1:
   command: /data/data/com.termux/files/usr/bin/python3 /data/data/com.termux/files/usr/lib/python3.9/site-packages/pip/_vendor/pep517/in_process/_in_process.py build_wheel /data/data/com.termux/files/usr/tmp/tmp1oys7jv4
       cwd: /data/data/com.termux/files/usr/tmp/pip-install-c2tb1lir/cryptography_c5bce2ef07b948b79c8dd32e9632c94f
  Complete output (165 lines):
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build/lib.linux-aarch64-3.9
  creating build/lib.linux-aarch64-3.9/cryptography
  copying src/cryptography/__about__.py -> build/lib.linux-aarch64-3.9/cryptography
  copying src/cryptography/__init__.py -> build/lib.linux-aarch64-3.9/cryptography
  copying src/cryptography/exceptions.py -> build/lib.linux-aarch64-3.9/cryptography
  copying src/cryptography/fernet.py -> build/lib.linux-aarch64-3.9/cryptography
  copying src/cryptography/utils.py -> build/lib.linux-aarch64-3.9/cryptography
  creating build/lib.linux-aarch64-3.9/cryptography/hazmat
  copying src/cryptography/hazmat/__init__.py -> build/lib.linux-aarch64-3.9/cryptography/hazmat
  copying src/cryptography/hazmat/_der.py -> build/lib.linux-aarch64-3.9/cryptography/hazmat
  copying src/cryptography/hazmat/_oid.py -> build/lib.linux-aarch64-3.9/cryptography/hazmat
  copying src/cryptography/hazmat/_types.py -> build/lib.linux-aarch64-3.9/cryptography/hazmat
  creating build/lib.linux-aarch64-3.9/cryptography/x509
  copying src/cryptography/x509/__init__.py -> build/lib.linux-aarch64-3.9/cryptography/x509
  copying src/cryptography/x509/base.py -> build/lib.linux-aarch64-3.9/cryptography/x509
  copying src/cryptography/x509/certificate_transparency.py -> build/lib.linux-aarch64-3.9/cryptography/x509
  copying src/cryptography/x509/extensions.py -> build/lib.linux-aarch64-3.9/cryptography/x509
  copying src/cryptography/x509/general_name.py -> build/lib.linux-aarch64-3.9/cryptography/x509
  copying src/cryptography/x509/name.py -> build/lib.linux-aarch64-3.9/cryptography/x509
  copying src/cryptography/x509/ocsp.py -> build/lib.linux-aarch64-3.9/cryptography/x509
  copying src/cryptography/x509/oid.py -> build/lib.linux-aarch64-3.9/cryptography/x509
  creating build/lib.linux-aarch64-3.9/cryptography/hazmat/backends
  copying src/cryptography/hazmat/backends/__init__.py -> build/lib.linux-aarch64-3.9/cryptography/hazmat/backends
  copying src/cryptography/hazmat/backends/interfaces.py -> build/lib.linux-aarch64-3.9/cryptography/hazmat/backends
  creating build/lib.linux-aarch64-3.9/cryptography/hazmat/bindings
  copying src/cryptography/hazmat/bindings/__init__.py -> build/lib.linux-aarch64-3.9/cryptography/hazmat/bindings
  creating build/lib.linux-aarch64-3.9/cryptography/hazmat/primitives
  copying src/cryptography/hazmat/primitives/__init__.py -> build/lib.linux-aarch64-3.9/cryptography/hazmat/primitives
  copying src/cryptography/hazmat/primitives/_asymmetric.py -> build/lib.linux-aarch64-3.9/cryptography/hazmat/primitives
  copying src/cryptography/hazmat/primitives/_cipheralgorithm.py -> build/lib.linux-aarch64-3.9/cryptography/hazmat/primitives
  copying src/cryptography/hazmat/primitives/_serialization.py -> build/lib.linux-aarch64-3.9/cryptography/hazmat/primitives
  copying src/cryptography/hazmat/primitives/cmac.py -> build/lib.linux-aarch64-3.9/cryptography/hazmat/primitives
  copying src/cryptography/hazmat/primitives/constant_time.py -> build/lib.linux-aarch64-3.9/cryptography/hazmat/primitives
  copying src/cryptography/hazmat/primitives/hashes.py -> build/lib.linux-aarch64-3.9/cryptography/hazmat/primitives
  copying src/cryptography/hazmat/primitives/hmac.py -> build/lib.linux-aarch64-3.9/cryptography/hazmat/primitives
  copying src/cryptography/hazmat/primitives/keywrap.py -> build/lib.linux-aarch64-3.9/cryptography/hazmat/primitives
  copying src/cryptography/hazmat/primitives/padding.py -> build/lib.linux-aarch64-3.9/cryptography/hazmat/primitives
  copying src/cryptography/hazmat/primitives/poly1305.py -> build/lib.linux-aarch64-3.9/cryptography/hazmat/primitives
  creating build/lib.linux-aarch64-3.9/cryptography/hazmat/backends/openssl
  copying src/cryptography/hazmat/backends/openssl/__init__.py -> build/lib.linux-aarch64-3.9/cryptography/hazmat/backends/openssl
  copying src/cryptography/hazmat/backends/openssl/aead.py -> build/lib.linux-aarch64-3.9/cryptography/hazmat/backends/openssl
  copying src/cryptography/hazmat/backends/openssl/backend.py -> build/lib.linux-aarch64-3.9/cryptography/hazmat/backends/openssl
  copying src/cryptography/hazmat/backends/openssl/ciphers.py -> build/lib.linux-aarch64-3.9/cryptography/hazmat/backends/openssl
  copying src/cryptography/hazmat/backends/openssl/cmac.py -> build/lib.linux-aarch64-3.9/cryptography/hazmat/backends/openssl
  copying src/cryptography/hazmat/backends/openssl/decode_asn1.py -> build/lib.linux-aarch64-3.9/cryptography/hazmat/backends/openssl
  copying src/cryptography/hazmat/backends/openssl/dh.py -> build/lib.linux-aarch64-3.9/cryptography/hazmat/backends/openssl
  copying src/cryptography/hazmat/backends/openssl/dsa.py -> build/lib.linux-aarch64-3.9/cryptography/hazmat/backends/openssl
  copying src/cryptography/hazmat/backends/openssl/ec.py -> build/lib.linux-aarch64-3.9/cryptography/hazmat/backends/openssl
  copying src/cryptography/hazmat/backends/openssl/ed25519.py -> build/lib.linux-aarch64-3.9/cryptography/hazmat/backends/openssl
  copying src/cryptography/hazmat/backends/openssl/ed448.py -> build/lib.linux-aarch64-3.9/cryptography/hazmat/backends/openssl
  copying src/cryptography/hazmat/backends/openssl/encode_asn1.py -> build/lib.linux-aarch64-3.9/cryptography/hazmat/backends/openssl
  copying src/cryptography/hazmat/backends/openssl/hashes.py -> build/lib.linux-aarch64-3.9/cryptography/hazmat/backends/openssl
  copying src/cryptography/hazmat/backends/openssl/hmac.py -> build/lib.linux-aarch64-3.9/cryptography/hazmat/backends/openssl
  copying src/cryptography/hazmat/backends/openssl/ocsp.py -> build/lib.linux-aarch64-3.9/cryptography/hazmat/backends/openssl
  copying src/cryptography/hazmat/backends/openssl/poly1305.py -> build/lib.linux-aarch64-3.9/cryptography/hazmat/backends/openssl
  copying src/cryptography/hazmat/backends/openssl/rsa.py -> build/lib.linux-aarch64-3.9/cryptography/hazmat/backends/openssl
  copying src/cryptography/hazmat/backends/openssl/utils.py -> build/lib.linux-aarch64-3.9/cryptography/hazmat/backends/openssl
  copying src/cryptography/hazmat/backends/openssl/x25519.py -> build/lib.linux-aarch64-3.9/cryptography/hazmat/backends/openssl
  copying src/cryptography/hazmat/backends/openssl/x448.py -> build/lib.linux-aarch64-3.9/cryptography/hazmat/backends/openssl
  copying src/cryptography/hazmat/backends/openssl/x509.py -> build/lib.linux-aarch64-3.9/cryptography/hazmat/backends/openssl
  creating build/lib.linux-aarch64-3.9/cryptography/hazmat/bindings/openssl
  copying src/cryptography/hazmat/bindings/openssl/__init__.py -> build/lib.linux-aarch64-3.9/cryptography/hazmat/bindings/openssl
  copying src/cryptography/hazmat/bindings/openssl/_conditional.py -> build/lib.linux-aarch64-3.9/cryptography/hazmat/bindings/openssl
  copying src/cryptography/hazmat/bindings/openssl/binding.py -> build/lib.linux-aarch64-3.9/cryptography/hazmat/bindings/openssl
  creating build/lib.linux-aarch64-3.9/cryptography/hazmat/primitives/asymmetric
  copying src/cryptography/hazmat/primitives/asymmetric/__init__.py -> build/lib.linux-aarch64-3.9/cryptography/hazmat/primitives/asymmetric
  copying src/cryptography/hazmat/primitives/asymmetric/dh.py -> build/lib.linux-aarch64-3.9/cryptography/hazmat/primitives/asymmetric
  copying src/cryptography/hazmat/primitives/asymmetric/dsa.py -> build/lib.linux-aarch64-3.9/cryptography/hazmat/primitives/asymmetric
  copying src/cryptography/hazmat/primitives/asymmetric/ec.py -> build/lib.linux-aarch64-3.9/cryptography/hazmat/primitives/asymmetric
  copying src/cryptography/hazmat/primitives/asymmetric/ed25519.py -> build/lib.linux-aarch64-3.9/cryptography/hazmat/primitives/asymmetric
  copying src/cryptography/hazmat/primitives/asymmetric/ed448.py -> build/lib.linux-aarch64-3.9/cryptography/hazmat/primitives/asymmetric
  copying src/cryptography/hazmat/primitives/asymmetric/padding.py -> build/lib.linux-aarch64-3.9/cryptography/hazmat/primitives/asymmetric
  copying src/cryptography/hazmat/primitives/asymmetric/rsa.py -> build/lib.linux-aarch64-3.9/cryptography/hazmat/primitives/asymmetric
  copying src/cryptography/hazmat/primitives/asymmetric/utils.py -> build/lib.linux-aarch64-3.9/cryptography/hazmat/primitives/asymmetric
  copying src/cryptography/hazmat/primitives/asymmetric/x25519.py -> build/lib.linux-aarch64-3.9/cryptography/hazmat/primitives/asymmetric
  copying src/cryptography/hazmat/primitives/asymmetric/x448.py -> build/lib.linux-aarch64-3.9/cryptography/hazmat/primitives/asymmetric
  creating build/lib.linux-aarch64-3.9/cryptography/hazmat/primitives/ciphers
  copying src/cryptography/hazmat/primitives/ciphers/__init__.py -> build/lib.linux-aarch64-3.9/cryptography/hazmat/primitives/ciphers
  copying src/cryptography/hazmat/primitives/ciphers/aead.py -> build/lib.linux-aarch64-3.9/cryptography/hazmat/primitives/ciphers
  copying src/cryptography/hazmat/primitives/ciphers/algorithms.py -> build/lib.linux-aarch64-3.9/cryptography/hazmat/primitives/ciphers
  copying src/cryptography/hazmat/primitives/ciphers/base.py -> build/lib.linux-aarch64-3.9/cryptography/hazmat/primitives/ciphers
  copying src/cryptography/hazmat/primitives/ciphers/modes.py -> build/lib.linux-aarch64-3.9/cryptography/hazmat/primitives/ciphers
  creating build/lib.linux-aarch64-3.9/cryptography/hazmat/primitives/kdf
  copying src/cryptography/hazmat/primitives/kdf/__init__.py -> build/lib.linux-aarch64-3.9/cryptography/hazmat/primitives/kdf
  copying src/cryptography/hazmat/primitives/kdf/concatkdf.py -> build/lib.linux-aarch64-3.9/cryptography/hazmat/primitives/kdf
  copying src/cryptography/hazmat/primitives/kdf/hkdf.py -> build/lib.linux-aarch64-3.9/cryptography/hazmat/primitives/kdf
  copying src/cryptography/hazmat/primitives/kdf/kbkdf.py -> build/lib.linux-aarch64-3.9/cryptography/hazmat/primitives/kdf
  copying src/cryptography/hazmat/primitives/kdf/pbkdf2.py -> build/lib.linux-aarch64-3.9/cryptography/hazmat/primitives/kdf
  copying src/cryptography/hazmat/primitives/kdf/scrypt.py -> build/lib.linux-aarch64-3.9/cryptography/hazmat/primitives/kdf
  copying src/cryptography/hazmat/primitives/kdf/x963kdf.py -> build/lib.linux-aarch64-3.9/cryptography/hazmat/primitives/kdf
  creating build/lib.linux-aarch64-3.9/cryptography/hazmat/primitives/serialization
  copying src/cryptography/hazmat/primitives/serialization/__init__.py -> build/lib.linux-aarch64-3.9/cryptography/hazmat/primitives/serialization
  copying src/cryptography/hazmat/primitives/serialization/base.py -> build/lib.linux-aarch64-3.9/cryptography/hazmat/primitives/serialization
  copying src/cryptography/hazmat/primitives/serialization/pkcs12.py -> build/lib.linux-aarch64-3.9/cryptography/hazmat/primitives/serialization
  copying src/cryptography/hazmat/primitives/serialization/pkcs7.py -> build/lib.linux-aarch64-3.9/cryptography/hazmat/primitives/serialization
  copying src/cryptography/hazmat/primitives/serialization/ssh.py -> build/lib.linux-aarch64-3.9/cryptography/hazmat/primitives/serialization
  creating build/lib.linux-aarch64-3.9/cryptography/hazmat/primitives/twofactor
  copying src/cryptography/hazmat/primitives/twofactor/__init__.py -> build/lib.linux-aarch64-3.9/cryptography/hazmat/primitives/twofactor
  copying src/cryptography/hazmat/primitives/twofactor/hotp.py -> build/lib.linux-aarch64-3.9/cryptography/hazmat/primitives/twofactor
  copying src/cryptography/hazmat/primitives/twofactor/totp.py -> build/lib.linux-aarch64-3.9/cryptography/hazmat/primitives/twofactor
  copying src/cryptography/hazmat/primitives/twofactor/utils.py -> build/lib.linux-aarch64-3.9/cryptography/hazmat/primitives/twofactor
  running egg_info
  writing src/cryptography.egg-info/PKG-INFO
  writing dependency_links to src/cryptography.egg-info/dependency_links.txt
  writing requirements to src/cryptography.egg-info/requires.txt
  writing top-level names to src/cryptography.egg-info/top_level.txt
  reading manifest file 'src/cryptography.egg-info/SOURCES.txt'
  reading manifest template 'MANIFEST.in'
  no previously-included directories found matching 'docs/_build'
  warning: no previously-included files found matching 'vectors'
  warning: no previously-included files matching '*' found under directory 'vectors'
  warning: no previously-included files matching '*' found under directory '.github'
  warning: no previously-included files found matching 'release.py'
  warning: no previously-included files found matching '.coveragerc'
  warning: no previously-included files found matching 'codecov.yml'
  warning: no previously-included files found matching '.readthedocs.yml'
  warning: no previously-included files found matching 'dev-requirements.txt'
  warning: no previously-included files found matching 'tox.ini'
  warning: no previously-included files found matching 'mypy.ini'
  warning: no previously-included files matching '*' found under directory '.zuul.d'
  warning: no previously-included files matching '*' found under directory '.zuul.playbooks'
  adding license file 'LICENSE'
  adding license file 'LICENSE.APACHE'
  adding license file 'LICENSE.BSD'
  adding license file 'LICENSE.PSF'
  writing manifest file 'src/cryptography.egg-info/SOURCES.txt'
  copying src/cryptography/py.typed -> build/lib.linux-aarch64-3.9/cryptography
  running build_ext
  generating cffi module 'build/temp.linux-aarch64-3.9/_padding.c'
  creating build/temp.linux-aarch64-3.9
  generating cffi module 'build/temp.linux-aarch64-3.9/_openssl.c'
  running build_rust

      =============================DEBUG ASSISTANCE=============================
      If you are seeing a compilation error please try the following steps to
      successfully install cryptography:
      1) Upgrade to the latest pip and try again. This will fix errors for most
         users. See: https://pip.pypa.io/en/stable/installing/#upgrading-pip
      2) Read https://cryptography.io/en/latest/installation.html for specific
         instructions for your platform.
      3) Check our frequently asked questions for more information:
         https://cryptography.io/en/latest/faq.html
      4) Ensure you have a recent Rust toolchain installed:
         https://cryptography.io/en/latest/installation.html#rust
      5) If you are experiencing issues with Rust for *this release only* you may
         set the environment variable `CRYPTOGRAPHY_DONT_BUILD_RUST=1`.
      =============================DEBUG ASSISTANCE=============================

  error: can't find Rust compiler

  If you are using an outdated pip version, it is possible a prebuilt wheel is available for this package but pip is not able to install from it. Installing from the wheel would avoid the need for a Rust compiler.

  To update pip, run:

      pip install --upgrade pip

  and then retry package installation.

  If you did intend to build this package from source, try installing a Rust compiler from your system package manager and ensure it is on the PATH during installation. Alternatively, rustup (available at https://rustup.rs) is the recommended way to download and update the Rust compiler toolchain.

  This package requires Rust >=1.41.0.
  ----------------------------------------
  ERROR: Failed building wheel for cryptography
Failed to build cryptography
ERROR: Could not build wheels for cryptography which use PEP 517 and cannot be installed directly

**Reply by SeaHOH** [reply]
看来还需要 Rust，安装即可。
```console
pkg rust
```

**Comment by chen2021-web** [comment]
运行最后一个命令时显示
mv: cannot stat 'start.py': No such file or directory

**Reply by SeaHOH** [reply]
因为 mv 这条命令只需要运行一次，第二次就会这样。

**Comment by chen2021-web** [comment]
然后我使用./start命令成功运行了

**Comment by chen2021-web** [comment]
但运行后显示
![Screenshot_20210809-191122](https://user-images.githubusercontent.com/83762621/128698086-a11066c3-c300-4c44-9d44-dcbc03b28591.jpg)

**Reply by chen2021-web** [reply]
请问怎么解决

**Reply by chen2021-web** [reply]
还有怎么给手机设置代理

**Reply by chen2021-web** [reply]
设置代理搞定了，但不知道要怎么安装证书

**Comment by chen2021-web** [comment]
请问在手机上要怎么安装GotoX的证书

**Reply by SeaHOH** [reply]
设置代理后浏览器访问 gotox.go

1. 浏览器直接点安装

1. 或点下载，然后设置里面搜索**加密与凭证**，选择安装证书

**Reply by SeaHOH** [reply]
提醒一点：扫描到足够到谷歌 IP 后 (用于直连访问谷歌服务)，请编辑设置禁用扫描，这个比较损耗内存。

浏览器访问 http://localhost/docmd?cmd=quit 是停止代理运行

**Comment by chen2021-web** [comment]
安装依赖项的时候报错ERROR: Command errored out with exit status 1:
   command: /data/data/com.termux/files/usr/bin/python3 /data/data/com.termux/files/usr/tmp/pip-standalone-pip-5diy_m8n/__env_pip__.zip/pip install --ignore-installed --no-user --prefix /data/data/com.termux/files/usr/tmp/pip-build-env-3ni83mot/overlay --no-warn-script-location --no-binary :none: --only-binary :none: -i https://pypi.org/simple -- 'setuptools >= 40.8.0' wheel 'Cython >= 3.0a9' 'cffi >= 1.12.3 ; platform_python_implementation == '"'"'CPython'"'"'' 'greenlet >= 0.4.17, < 2.0 ; platform_python_implementation == '"'"'CPython'"'"''
       cwd: None
  Complete output (111 lines):
  Collecting setuptools>=40.8.0
    Using cached setuptools-60.1.0-py3-none-any.whl (952 kB)
  Collecting wheel
    Using cached wheel-0.37.1-py2.py3-none-any.whl (35 kB)
  Collecting Cython>=3.0a9
    Using cached Cython-3.0.0a9-py2.py3-none-any.whl (1.1 MB)
  Collecting cffi>=1.12.3
    Using cached cffi-1.15.0.tar.gz (484 kB)
    Preparing metadata (setup.py): started
    Preparing metadata (setup.py): finished with status 'done'
  Collecting greenlet<2.0,>=0.4.17
    Using cached greenlet-1.1.2.tar.gz (91 kB)
    Preparing metadata (setup.py): started
    Preparing metadata (setup.py): finished with status 'done'
  Collecting pycparser
    Using cached pycparser-2.21-py2.py3-none-any.whl (118 kB)
  Using legacy 'setup.py install' for cffi, since package 'wheel' is not installed.
  Using legacy 'setup.py install' for greenlet, since package 'wheel' is not installed.
  Installing collected packages: pycparser, wheel, setuptools, greenlet, Cython, cffi
      Running setup.py install for greenlet: started
      Running setup.py install for greenlet: finished with status 'error'
      ERROR: Command errored out with exit status 1:
       command: /data/data/com.termux/files/usr/bin/python3 -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/data/data/com.termux/files/usr/tmp/pip-install-_4343_yo/greenlet_dbc459e367d5443a8587e2ee5eb0f671/setup.py'"'"'; __file__='"'"'/data/data/com.termux/files/usr/tmp/pip-install-_4343_yo/greenlet_dbc459e367d5443a8587e2ee5eb0f671/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /data/data/com.termux/files/usr/tmp/pip-record-y56009ck/install-record.txt --single-version-externally-managed --prefix /data/data/com.termux/files/usr/tmp/pip-build-env-3ni83mot/overlay --compile --install-headers /data/data/com.termux/files/usr/tmp/pip-build-env-3ni83mot/overlay/include/python3.10/greenlet
           cwd: /data/data/com.termux/files/usr/tmp/pip-install-_4343_yo/greenlet_dbc459e367d5443a8587e2ee5eb0f671/
      Complete output (84 lines):
      running install
      running build
      running build_py
      creating build
      creating build/lib.linux-aarch64-3.10
      creating build/lib.linux-aarch64-3.10/greenlet
      copying src/greenlet/__init__.py -> build/lib.linux-aarch64-3.10/greenlet
      creating build/lib.linux-aarch64-3.10/greenlet/tests
      copying src/greenlet/tests/__init__.py -> build/lib.linux-aarch64-3.10/greenlet/tests
      copying src/greenlet/tests/test_contextvars.py -> build/lib.linux-aarch64-3.10/greenlet/tests
      copying src/greenlet/tests/test_cpp.py -> build/lib.linux-aarch64-3.10/greenlet/tests
      copying src/greenlet/tests/test_extension_interface.py -> build/lib.linux-aarch64-3.10/greenlet/tests
      copying src/greenlet/tests/test_gc.py -> build/lib.linux-aarch64-3.10/greenlet/tests
      copying src/greenlet/tests/test_generator.py -> build/lib.linux-aarch64-3.10/greenlet/tests
      copying src/greenlet/tests/test_generator_nested.py -> build/lib.linux-aarch64-3.10/greenlet/tests
      copying src/greenlet/tests/test_greenlet.py -> build/lib.linux-aarch64-3.10/greenlet/tests
      copying src/greenlet/tests/test_leaks.py -> build/lib.linux-aarch64-3.10/greenlet/tests
      copying src/greenlet/tests/test_stack_saved.py -> build/lib.linux-aarch64-3.10/greenlet/tests
      copying src/greenlet/tests/test_throw.py -> build/lib.linux-aarch64-3.10/greenlet/tests
      copying src/greenlet/tests/test_tracing.py -> build/lib.linux-aarch64-3.10/greenlet/tests
      copying src/greenlet/tests/test_version.py -> build/lib.linux-aarch64-3.10/greenlet/tests
      copying src/greenlet/tests/test_weakref.py -> build/lib.linux-aarch64-3.10/greenlet/tests
      running egg_info
      writing src/greenlet.egg-info/PKG-INFO
      writing dependency_links to src/greenlet.egg-info/dependency_links.txt
      writing requirements to src/greenlet.egg-info/requires.txt
      writing top-level names to src/greenlet.egg-info/top_level.txt
      reading manifest file 'src/greenlet.egg-info/SOURCES.txt'
      reading manifest template 'MANIFEST.in'
      no previously-included dire

**Reply by SeaHOH** [reply]
这个错误日志不完整。

**Reply by SeaHOH** [reply]
安装前需要更新组件，前面的命令已经修改。
